controllers/ocsinitialization/ocsinitialization_controller.go

```diff
@@ -171,6 +180,23 @@ func (r *OCSInitializationReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&ocsv1.OCSInitialization{}).
 		Owns(&appsv1.Deployment{}).
+		Watches(&source.Kind{Type: &configv1.ClusterVersion{}}, handler.EnqueueRequestsFromMapFunc(func(obj client.Object) []reconcile.Request {
```

Comment:
@malayparida2000 is there a need for a watcher here??

Reply:
If somehow the clusterversion CR changes(someone gives a new name for the cluster ID, We should catch that), Thus the watcher

Reply:
I dont think that's the case; the clusterID will always remain the same for the entire cluster lifecycle. as its an OCP cluster-ID.

Reply:
The spec.clusterID field for a clusterversion cr is actually editable, So someone can actually edit it if they want.